ci.rb

```diff
@@ -335,12 +335,15 @@ def get_limits(t_options, t_client, t_repo)
                   regression_baselines << [compiler, regression_base];
 
                   if !File.directory?(regression_base.get_build_dir(compiler))
```

Comment:
This (!File.directory) means that the folder is NOT there. So removing (rm) it on line 339 will not change anything.

You have to remove the folder before you get here.

                  if File.directory?(regression_base.get_build_dir(compiler))
                    $logger.info "Removing pre-existing regression build directory (#{regression_base.get_build_dir(compiler)})"
                    system("rm -rf #{regression_base.get_build_dir(compiler)}")
                  end

Reply:
So I suggest moving the new lines down into the else and bracketing with if(keep_build_folder) print "skipping" else remove folder.

Reply:
...else remove folder and do lines 340-342.

Reply:
This PR became a **crazy** overhaul of Decent CI.  Much slimmer now, and a new build folder setup which will enable ccache capability to speed up builds.  Folders should be cleaned up nicely between every compiler run now.  You can check out the new process here: https://github.com/Myoldmopar/decent_ci/blob/d08606fdb7434b8fc56b75c5cb4fc468d62bbeb4/ci.rb#L274-L316